src/net/encoding.c

```diff
@@ -39,7 +39,11 @@ _z_encoding_t _z_encoding_wrap(uint16_t id, const char *schema) {
 
 _z_encoding_t _z_encoding_null(void) { return _z_encoding_wrap(_Z_ENCODING_ID_DEFAULT, NULL); }
 
-void _z_encoding_clear(_z_encoding_t *encoding) { _z_string_clear(&encoding->schema); }
+void _z_encoding_clear(_z_encoding_t *encoding) {
+    if (_z_string_check(&encoding->schema)) {
```

Comment:
_z_string_check does not seem to be necessary here. If it is false, _z_string_clear will be a no-op